examples/misc/compare-with-kie-editors-standalone/index.html

```diff
@@ -0,0 +1,317 @@
+      logBpmnVisu('Diagram already loaded, skipping new load');
+      return;
+    }
+    logBpmnVisu('Start loading diagram');
+    bpmnVisualization.load(xml, { fit: computeBpmnVisualizationFitOptions() });
+    logBpmnVisu('Diagram loaded');
+    isBpmnVisualizationAlreadyLoaded = true;
+  }
+
+  async function loadWithKieBpmn(xml) {
+    if (isKieBpmnAlreadyLoaded) {
+      logKieBpmn('Diagram already loaded, skipping new load');
+      return;
+    }
+    logKieBpmn('Starting setting content')
+    // TODO clarify if setContent returns a promise
+    // The introduction article and 0.8.1 readme states it returns 'void', but the abstract class TypeScript definition
+    // declares a Promise
+    // Using promise doesn't change anything, the load is fully asynchronous. We may have to register event listener
+    // see https://github.com/kiegroup/kogito-tooling/blob/e151ccdac6ae526e81dc5e0ec302f02c922b00ea/packages/editor/src/api/Editor.ts#L36
+    // kieBpmnEditor.setContent(xml)
+    //         .then(value => logKieBpmn('Content set'))
+    //         .then(value => setZoomLevelKieBpmn())
+    //         .catch(reason => console.error('[KIE]', reason));
+    kieBpmnEditor.setContent(xml);
+    logKieBpmn('Content set');
+    setZoomLevelKieBpmn();
+    isKieBpmnAlreadyLoaded = true;
+  }
+
+  function setZoomLevelKieBpmn() {
+    // TODO see if we have something for kie-editors-standalone
+    logKieBpmn('no zoom settings for now');
+  }
+
+  function setZoomLevelBpmnVisualization() {
+    bpmnVisualization.fit(computeBpmnVisualizationFitOptions());
+    logBpmnVisu(`Zoom level set, fitView ${fitView}`);
+  }
+
+  function computeBpmnVisualizationFitOptions() {
+    const fitType = fitView ? bpmnvisu.FitType.Center: bpmnvisu.FitType.None;
+    return { type: fitType, margin: 10 };
+  }
+
+  function handleFileSelect(evt) {
+    evt.stopPropagation();
+    evt.preventDefault();
+
+    const file = evt.target.files[0];
+    logMain(`New file selected for load: ${file.name}`);
+    const reader = new FileReader();
+    reader.onload = () => {
+      isBpmnVisualizationAlreadyLoaded = false;
+      isKieBpmnAlreadyLoaded = false;
+      loadBpmn(reader.result);
+
+      document.getElementById('loading-info').classList.remove('hidden');
+      document.querySelector('#loading-info > span').innerHTML = `<code>${file.name}</code>`;
+    };
+    reader.readAsText(file);
+  }
+
+  function loadBpmn(xml) {
+    const useBpmnVisualization = isUseBpmnVisualization();
+    displayBpmnContainer(useBpmnVisualization);
+
+    // effective load
+    try {
+      logMain(`Ready to load BPMN with fitView: ${fitView}`);
+      if (useBpmnVisualization) {
+        loadWithBpmnVisualization(xml);
+      } else {
+        loadWithKieBpmn(xml);
+      }
+      loadedXml = xml;
+    } catch (err) {
+      console.error('[main] Unable to load BPMN diagram', err);
+    }
+  }
+
+  function reloadAfterBpmnLibrarySwitch() {
```

Comment:
Maybe we can extract the common methods with _compare-with-bpmn-js_ to a file utils 🙂

Reply:
Yes, there are duplications 😸 
I can try something in a new refactoring PR

Reply:
See #127